src/EditorFeatures/VisualBasicTest/TypeInferrer/TypeInferrerTests.vb

```diff
@@ -684,7 +684,7 @@ End Class
         <WorkItem(994388, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/994388")>
         Public Async Function TestCatchFilterClause2() As Task
             Dim text = "Try : Catch ex As Exception When [|foo|].N"
-            Await TestInMethodAsync(text, "System.Object", testPosition:=False)
+            Await TestInMethodAsync(text, "System.Boolean", testPosition:=False)
```

Comment:
But you're not the expression of the "when clause".  "foo.N" is the expression of the "When clause".   

Note the type inferrer works in two ways.  One is node based and one is position based.  The position after the "When" is definitely a boolean context.  But you have a node here.